runtime/rastrace/method_trace.c

```diff
@@ -477,10 +477,19 @@ traceMethodArgObject(J9VMThread *thr, UDATA* arg0EA, char* cursor, UDATA length)
 		J9Class* clazz = J9OBJECT_CLAZZ(thr, object);
 		J9ROMClass * romClass = clazz->romClass;
 		J9UTF8* className = J9ROMCLASS_CLASSNAME(romClass);
+		J9JavaVM *vm = thr->javaVM;
 
-		/* TODO: handle arrays */
+		if (clazz == J9VMJAVALANGSTRING_OR_NULL(vm)) {
+			/* string arg */
+			char stringArgBuffer[J9_ARRAY_DIMENSION_LIMIT];
 
-		j9str_printf(PORTLIB, cursor, length, "%.*s@%p", (U_32)J9UTF8_LENGTH(className), J9UTF8_DATA(className), object);
+			J9InternalVMFunctions const * const vmFuncs = thr->javaVM->internalVMFunctions;
+			char *stringArgUTF8 = vmFuncs->copyStringToUTF8WithMemAlloc(thr, object, J9_STR_NULL_TERMINATE_RESULT, "  ", 2, stringArgBuffer, J9_ARRAY_DIMENSION_LIMIT, NULL);
+			j9str_printf(PORTLIB, cursor, length, "%.*s@%.*s", (U_32)J9UTF8_LENGTH(className), J9UTF8_DATA(className), J9UTF8_LENGTH(stringArgUTF8), J9UTF8_DATA(stringArgUTF8));
```

Comment:
if you hit the string length limit, you want to add a `...` so its clear to users that the string is truncated

Reply:
Also, I dont think you need the `(U_32)J9UTF8_LENGTH(className), J9UTF8_DATA(className),` 
since in this case you already know its a string. You could just do "(String)%.*s".